commands/build.go

```diff
@@ -643,6 +653,81 @@ func parsePrintFunc(str string) (*build.PrintFunc, error) {
 	return f, nil
 }
 
+func addGitProvenance(labels []string, contextPath string, dockerfilePath string) ([]string, error) {
+	v, ok := os.LookupEnv("BUILDX_GIT_LABELS")
+	if !ok || contextPath == "" {
+		return labels, nil
+	}
+
+	if len(labels) == 0 {
+		labels = make([]string, 0)
+	}
+
+	// figure out in which directory the git command needs to run
+	var wd string
+	if filepath.IsAbs(contextPath) {
+		wd = contextPath
+	} else {
+		cwd, _ := os.Getwd()
+		wd, _ = filepath.Abs(filepath.Join(cwd, contextPath))
+	}
+
+	// obtain Git sha of current HEAD
+	cmd := exec.Command("git", "rev-parse", "HEAD")
+	cmd.Dir = wd
+	sha, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("error obtaining git head: %w", err)
+	}
+
+	// check if the current HEAD is clean
+	cmd = exec.Command("git", "diff", "--quiet", "--exit-code")
+	cmd.Dir = wd
+	_, err = cmd.Output()
+	if err == nil {
+		labels = append(labels, []string{fmt.Sprintf("%s=%s", ImageRevisionLabel, strings.TrimSpace(string(sha)))}...)
+	} else {
+		labels = append(labels, []string{fmt.Sprintf("%s=%s-dirty", ImageRevisionLabel, strings.TrimSpace(string(sha)))}...)
+	}
+
+	// add the origin url if full Git details are requested
+	if v == "full" {
+		cmd = exec.Command("git", "config", "--get", "remote.origin.url")
+		cmd.Dir = wd
+		remote, err := cmd.Output()
+		if err != nil {
+			return nil, fmt.Errorf("failed to obtain remote.origin.url: %w", err)
+		}
+		labels = append(labels, []string{fmt.Sprintf("%s=%s", ImageSourceLabel, strings.TrimSpace(string(remote)))}...)
+	}
+
+	// add Dockerfile path; there is no org.opencontainers annotation for this
+	if dockerfilePath == "" {
+		dockerfilePath = "Dockerfile"
+	}
+
+	// obtain Git root directory
+	cmd = exec.Command("git", "rev-parse", "--show-toplevel")
+	cmd.Dir = wd
+	root, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+
+	// make the Dockerfile path relative to the Git root
+	if !filepath.IsAbs(dockerfilePath) {
+		cwd, _ := os.Getwd()
+		dockerfilePath = filepath.Join(cwd, dockerfilePath)
+	}
+	dockerfilePath, err = filepath.Rel(strings.TrimSpace(string(root)), dockerfilePath)
+	if err != nil {
+		return nil, err
```

Comment:
Dockerfile path does not need to be relative to the context directory. This should not fail the build. And path should never be `../../smth`

Reply:
Fixed!